pkg/trino/driver/driver.go

```
@@ -48,6 +57,19 @@ func Open(settings models.TrinoDatasourceSettings) (*sql.DB, error) {
 			},
 		},
 	}
+	if settings.TokenUrl != "" && settings.ClientId != "" && settings.ClientSecret != "" {
```

If all these fields are required, we should throw an error if any one of them is set, but any other one is empty.

---

My idea was if any of them set then use a new approach for client credentials flow. If any of them is not set then just skip and an old appoach is used. I just don't want to break something for anyone

---

All these fields are new, so there's no existing behavior to break. There's a chance a user will not fill in one of these fields, for example when copying and pasting values from their IDP, and if we won't even attempt using them in such case, users might spend a lot of time trying to debug it. Being strict here makes the behavior more predictable.

---

So your suggestion  is to throw exception in case if one or more of these fields are set but not all?

---

Yes

---

fixed